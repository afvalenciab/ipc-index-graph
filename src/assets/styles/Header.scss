.header {
  display: flex;
  box-sizing: border-box;
  justify-content: space-between;
  align-items: center;
  padding: 15px 50px;
  box-shadow: 0 3px 7px -5px rgba(0,0,0,.6);
  
  a {
    text-decoration: none;
    color: #000;
  }

  h1 {
    margin: 0;
    font-weight: 800;
    letter-spacing: 5px;

    span {
      margin-left: 8px;
      font-weight: 300;
      letter-spacing: normal;
    }
  }

  .menu {
    ul {
      margin: 0;
      padding: 0;
      display: flex;
      flex-wrap: wrap;
      justify-content: flex-end;
      align-items: center;
      list-style: none;

      li {
        margin: 0;
        margin-left: 10px ;

        a {
          display: inline-block;
          text-decoration: none;
          font-weight: 600;
          color: #000;
          transition: 1s;
          will-change: opacity, transform;
        }
      }

      li:hover a {
        transform: translateY(-5px);
        opacity: .6;
      }

      .menu--register {
        padding: 0;
        a {
          padding: 8px;
          background: #000;
          border-radius: 50px;
          color: #FFF;
          font-weight: 300;
        }
      }
    }
  }
}


@media screen and (max-width: 480px) {
  .header {
    padding: 15px 20px;
  
    h1 {
      span {
        margin-left: 5px;
      }
    }

    .menu {
      font-size: 14px;
      ul {
        flex-wrap: wrap;
        justify-content: flex-end;
        align-items: center;

        .menu--register {
          a {
            padding: 5px;
          }
        }
      }
    }
  }
}

@media screen and (max-width: 390px) {
  .header {
    padding: 15px 20px;

    h1 {
      font-size: 24px; 
      span {
        margin-left: 5px;
      }
    }
  }
}

@media screen and (max-width: 331px) {
  .header {
    padding: 15px 15px;
  }
}