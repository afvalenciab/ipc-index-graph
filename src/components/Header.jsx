import React from 'react';
import { Link } from 'react-router-dom';
import '../assets/styles/Header.scss';

const Header = () => {
  return (
    <header className='header'>
      <h1>
        IPC
        <span>Index</span>
      </h1>
      <nav className='menu'>
        <ul>
          <li>
            <Link to='/'>Administrar</Link>
          </li>
          <li>
            <Link to='/login'>Ingresar</Link>
          </li>
          <li className='menu--register'>
            <Link to='/register'>Crear cuenta</Link>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
