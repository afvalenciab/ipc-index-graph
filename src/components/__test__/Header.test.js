import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMocks';
import Header from '../Header';

describe('Header testing', () => {
  test('Match Snapshot', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
