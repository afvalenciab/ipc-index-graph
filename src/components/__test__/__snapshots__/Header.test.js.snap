// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header testing Match Snapshot 1`] = `
<header
  className="header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      IPC
      <span>
        Index
      </span>
    </h1>
  </a>
  <nav
    className="menu"
  >
    <ul>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Ingresar
        </a>
      </li>
      <li
        className="menu--register"
      >
        <a
          href="/register"
          onClick={[Function]}
        >
          Crear cuenta
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
